scarb/src/ops/new.rs

```diff
@@ -75,18 +76,37 @@ pub fn init_package(opts: InitOptions, config: &Config) -> Result<NewResult> {
 }
 
 fn infer_name(name: Option<PackageName>, path: &Utf8Path) -> Result<PackageName> {
-    if let Some(name) = name {
-        Ok(name)
+    let name = if let Some(name) = name {
+        name
     } else {
         let Some(file_name) = path.file_name() else {
             bail!(formatdoc! {r#"
                 cannot infer package name from path: {path}
                 help: use --name to override
             "#});
         };
+        PackageName::try_new(file_name)?
+    };
+
+    if PackageName::is_internal(&name) {
+        warn!(
+            "The name `{name}` is a Scarb internal package, \
+            it is recommended to use a different name to avoid problems"
+        )
+    }
 
-        PackageName::try_new(file_name)
+    if PackageName::is_windows_restricted(&name) {
+        if cfg!(windows) {
+            bail!("the name `{name}` cannot be used on Windows, it is a Windows reserved filename");
```

Comment:
kinda repeating wording here

```suggestion
            bail!("cannot use name `{name}`, it is a Windows reserved filename");
```